src/time.rs

```diff
@@ -124,3 +124,7 @@ impl Instant {
         DWT::get_cycle_count().wrapping_sub(self.now)
     }
 }
+
+/// Time unit
+#[derive(PartialEq, PartialOrd, Clone, Copy)]
```

Comment:
I think this should be expanded to a merge general time type that can be used in the system, thoughts on this?
Maybe there exists a crate for this as well already?

Reply:
Missed that we have the `Hertz` type, I'd recommend to maybe use that. 

Or expand with a new time type that is a bit more generic and fits the existing `U32Ext` trait.
Looking at the Hertz implementation the smallest quanta is 1 Hz which does not give much granularity.
It could make sense to have a new Time type, as you have started, but generalizing.

Reply:
Using `hertz` would also limit any function to <=1000ms.
I could expand `U32Ext` to incorporate `MilliSeconds` and change the functions to take that as parameter instead of u32.

> I think this should be expanded to a merge general time type that can be used in the system, thoughts on this?

Are you thinking of something like what is in `datetime.rs`? A time system with micros, millis, seconds, minutes and hours, and convertions in between them?

Reply:
I have expanded `U32Ext` for now.